hugolib/site.go

```diff
@@ -1431,27 +1432,47 @@ func (s *Site) assembleMenus() {
 	}
 
 	sectionPagesMenu := s.Info.sectionPagesMenu
-	sectionPagesMenus := make(map[string]interface{})
-	//creating flat hash
 	pages := s.Pages
-	for _, p := range pages {
 
-		if sectionPagesMenu != "" {
+	if sectionPagesMenu != "" {
+		// Create menu entries for section pages with content
+		for _, p := range pages {
+			if p.Section() != "" && p.Kind == KindSection {
```

Comment:
Oh, sorry, in this case you are right, on the second pass is when this test is supposed to be relevant.